content/en/docs/reference/kubectl/jsonpath.md

```diff
@@ -98,4 +98,17 @@ kubectl get pods -o=jsonpath="{range .items[*]}{.metadata.name}{\"\t\"}{.status.
 ```
 {{< /note >}}
 
+{{< note >}}
+
+JSONPath regular expressions are not supported. If you want to match using regular expressions, you can use a tool such as `jq`.
+
+```shell
+# kubectl does not suport regular expressions
+# The following command does not work
+kubectl get pods -o jsonpath='{.items[?(@.metadata.name=~/^test$/)].metadata.name}'
```

Comment:
Hi, The problem is that JSONPath is broken on kubectl, that is the whole point of this pull request. It does not support regular expressions. Developers like myself that have opened hundreds of issues on stackoverflow or kubernetes issues expecting regular expressions to work only to find out that kubectl has an incomplete JSONPath implementation. as per https://github.com/kubernetes/kubernetes/issues/72220 per @lavalamp suggestion I opened this PR.

That kubectl command does not exist on the cheatsheet because it does not run. Gives a runtime error

`error: error parsing jsonpath {.items[?(@.metadata.name=~/^test$/)].metadata.name}, unrecognized character in action: U+007E '~'
`
The examples have been added per request on this PR. So please decide and let me know how to proceed, in regards to the examples. The only thing that matters is that developers get explicitly warned that regular expressions are not supported in Kubectl JSONPath commands.

Reply:
“broken” is perhaps a bit strong given there isn't a JSONpath specification for kubectl (not) to conform too.

Point taken though, I'm happy to see an example of a command that readers might expect to work but actually doesn't.

Reply:
Well, to me it is broken, since it does not provide a useful error message and does not fail gracefully, you simply get that error message and are left to search why its not working, since I work for many years with JSONPath in different context the expectation is that regular expressions work. 

As a developer it is quite confusing, you think the problem is with your regex, you test it, it works, you try again, check the spaces, format, etc... and it does not work in kubectl. As you can see it can be quite frustrating and since the PR made to have go style regex has been refused, there is not much else to improve the current situation than to document it and avoid others to be frustrated.

In the example, readers expect to have as output all the pods with test in the name. But instead they get an error that is not very useful. seems like a unicode issue, but its not.

The bigger question is if JSONPath should be used at all. If it should not be replaced by something else that is less open to interpretation.

Reply:
@webmutation, Thanks for the explanation and adding the note.
From the first sentence, it is clear that regular expressions are not supported in kubectl.
The kubectl Cheat sheet is probably a good place to list working examples and helpful hints.
/lgtm

Reply:
Yeah I wouldn't use the word "broken" since it's use case dependent. But the current proposed change looks accurate and useful.